### PR TITLE
Implement get, update, add and remove for base repository

### PR DESCRIPTION
**Pull request by TAVMWB**

### CODE REVIEW

File: src/holon/rule_engine/repositories/repository_base.py [hunk 1]
```diff
@@ -111,43 +105,46 @@ def get_subset_range(
...
+    def add(self, object: PolymorphicModel):
+        """Add an object to the repository"""
+
+        # check if object base type is correct
+        if not self.base_model_type.__name__ == utils.get_base_polymorphic_model(object).__name__:
+            raise ValueError(
+                f"Can only insert objects of type {self.base_model_type.__name__}. Object type for attempted insertion: {utils.get_base_polymorphic_model(object).__name__}"
+            )
+
+        self.objects.append(object)
```
Comment (SCHS13) [inline src/holon/rule_engine/repositories/repository_base.py:142]: id's for added object need to be unique/changed

File: src/holon/rule_engine/repositories/repository_base.py [hunk 2]
```diff
@@ -111,43 +105,46 @@ def get_subset_range(
...
         return self.__class__(objects)
 
-    # TODO ERIK
-    def get(self, id: int) -> object:
-        """Get an item in the objects list by id"""
+    def get(self, id: int) -> PolymorphicModel:
+        """Get an item in the objects list by id. Return None if object was not found"""
 
-        raise NotImplementedError()
+        return next((object for object in self.objects if object.id == id), None)
```
[the remark this answers is not in the document]
Comment (TAVMWB) [reply src/holon/rule_engine/repositories/repository_base.py:111]: Good point, let's keep the behaviour the same as Django as much as possible. I'll push a correction soon

File: src/holon/rule_engine/repositories/repository_base.py (hunk 1 above)
Comment (Erikvv) [reply src/holon/rule_engine/repositories/repository_base.py:142]: The caller would probably like to know the id. So either return the object here, or let the caller set the id beforehand.

In either case we can have a global atomic counter which can create new id's.

I'm in favor of always using UUID's and never integer ids so id creation doesn't have to be coordinated. But that would be a different change.

Comment (TAVMWB) [reply src/holon/rule_engine/repositories/repository_base.py]: I made an id counter that starts at the max id of the provided objects + 1 and increments for every add action.

[the remark this answers is not in the document]
Comment (TAVMWB) [reply src/holon/rule_engine/repositories/repository_base.py]: done, `copy.deepcopy` seems to do the trick here

**Review by SCHS13 — APPROVED** [review]

**Pull request merged**